lib/u3d_core/core_ext/hash.rb

```diff
@@ -0,0 +1,30 @@
+## --- BEGIN LICENSE BLOCK ---
+# Copyright (c) 2017-present WeWantToKnow AS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+## --- END LICENSE BLOCK ---
+module CoreExtensions
+  module Extractable
+    refine Hash do
+      def extract(*keys)
+        select { |k, _v| keys.include?(k) }
```

Comment:
```ruby
def extract(*args)
  self.keys & args
end
```
This may be cleaner?

Reply:
does that work?

Reply:
Oops, maybe not. This should though:
```ruby
def extract(*args)
  self.keys & args
end
```

Reply:
![image](https://user-images.githubusercontent.com/24282/34665552-ebc0fc54-f460-11e7-9f61-96ca4ab39cd1.png)

Reply:
Misunderstood the intent. Sorry